crates/ra_ide_api_light/src/join_lines.rs

```diff
@@ -120,11 +120,35 @@ fn remove_newline(
     }
 }
 
+fn has_comma_after(node: &SyntaxNode) -> bool {
+    let next = node.next_sibling();
+    let nnext = node.next_sibling().and_then(|n| n.next_sibling());
+
+    match (next, nnext) {
+        // Whitespace followed by a comma is fine
+        (Some(ws), Some(comma)) if ws.kind() == WHITESPACE && comma.kind() == COMMA => true,
+        (Some(n), _) => n.kind() == COMMA,
+        _ => false,
+    }
+}
```

Comment:
I've just remembered that we already have something similar:

https://github.com/rust-analyzer/rust-analyzer/blob/55da8e786d2cac3cb6332f80c1ee715fbcbdee0d/crates/ra_assists/src/lib.rs#L107

Let's move that function to `ra_syntax::algo` and use it here?

Reply:
Sounds good, will do.